src/hotspot/share/opto/compile.cpp

```diff
@@ -3412,6 +3412,21 @@ void Compile::final_graph_reshaping_main_switch(Node* n, Final_Reshape_Counts& f
     break;
   }
 
+  case Op_ReverseBytesV:
+  case Op_ReverseV: {
```

Comment:
Can you elaborate, please, why it is performed so late in the optimization phase (at the very end during graph reshaping) and not during GVN?

Reply:
Its more of a chicken-egg problem here, for masked reverse operation, Reverse IR node is followed by a Blend Node, thus in such a case doing an eager Identity transform in Reverse::Identity will not work, also deferring this to blend may also not work since it could be a non-masked reverse operation, we could have handled it as a special case in inline_vector_nary_operation, but handling such special case in final graph reshaping looked more appropriate.

https://github.com/openjdk/panama-vector/pull/182#discussion_r845678080

Reply:
Do you mean it's important to apply the transformation at the right node (pick the right node as the root) and it is hard to make a decision during GVN?

Reply:
Yes, that what I meant.

Reply:
Addressed this after internally discussing with Sandhya. Moved the transforms from final graph re-shaping back to vector intrinsic routines.